app/models/action_page.rb

```diff
@@ -36,6 +54,24 @@ class ActionPage < ActiveRecord::Base
 
   scope :categorized, ->(category) { joins(:category).where(categories: { title: category }) }
 
+  def self.type(*types)
+    scopes = Array(types).flatten.map do |t|
+      unless %w(call congress_message email petition tweet redirect).include?(t)
+        raise ArgumentError, "unrecognized type #{t}"
+      end
+
+      where(:"enable_#{t}" => true)
+    end
+
+    seed = scopes.shift || all
```

Comment:
It looks like `inject`/`reduce` can be [called without a seed](https://apidock.com/ruby/Enumerable/inject) - could 66-74 be replaced with
```
    return all if types.empty?
    scopes.inject{ |scope, cond| scope.or(cond) }
```
?

Reply:
Ooh cool. `scopes.inject(:or)` also works :D